gcov: failed to enable coverage for qemu_cortex_a53

**Describe the bug**
Build error when enabling COVERAGE for qemu_cortex_a53 (arm64)

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? qemu_cortex_a53
 - What have you tried to diagnose or workaround this issue?  OK to enable COVERAGE for qemu_arc_em and qemu_riscv64

**To Reproduce**
Steps to reproduce the behavior:
1. Insert a line at [Kconfig.board:L9](https://github.com/zephyrproject-rtos/zephyr/blob/zephyr-v3.5.0/boards/arm64/qemu_cortex_a53/Kconfig.board#L9):
       select HAS_COVERAGE_SUPPORT
2. west -v build -b qemu_cortex_a53  zephyr/samples/hello_world -p -t run -- -DCONFIG_COVERAGE=y -DCONFIG_COVERAGE_GCOV=y -DCONFIG_COVERAGE_DUMP=y
3. See error
**undefined symbol `__gcov_bss_end'**
```
[98/104] : && ccache /opt/zephyr-sdk-0.16.1/aarch64-zephyr-elf/bin/aarch64-zephyr-elf-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  
  :
  :
cd /home/archilisW/zephyr_350/out/qemu_cortex_a53/hello_world/zephyr && /usr/bin/cmake -E true
/opt/toolchains/zephyr-sdk-0.16.1/aarch64-zephyr-elf/bin/../lib/gcc/aarch64-zephyr-elf/12.2.0/../../../../aarch64-zephyr-elf/bin/ld.bfd:
zephyr/linker_zephyr_pre0.cmd:181: undefined symbol `__gcov_bss_end' referenced in expression
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Expected behavior**
Build pass and a similar result as what we get with qemu_arc_em:
```
*** Booting Zephyr OS build 6b58156c36c9 ***
Hello World! qemu_arc

GCOV_COVERAGE_DUMP_START
*/home/archilisW/zephyr_350/out/qemu_arc_em/hello_world/zephyr/kernel/CMakeFiles/kernel.dir/work.c.gcda< 61 64 63 ......
```

**Impact**
Unable to use gcov for our arm64 platform

**Environment (please complete the following information):**
 - OS: Linux    (14.04.1-Ubuntu)
 - Docker Image: [v0.26.4](https://github.com/zephyrproject-rtos/docker-image/pkgs/container/zephyr-build/92729476?tag=v0.26.4)
 - Toolchain : zephyr-sdk-0.16.1
 - CMake version: 3.22.1
 - Python3: 3.10.6
 - Commit SHA or Version used:  zephyr-v3.5.0

--- a/subsys/testsuite/coverage/coverage_ram.ld
+++ b/subsys/testsuite/coverage/coverage_ram.ld
@@ -6,7 +6,7 @@
 
 /* Copied from linker.ld */
 
-#ifdef CONFIG_ARM
+#if defined(CONFIG_ARM) || defined(CONFIG_ARM64)
 SECTION_DATA_PROLOGUE(_GCOV_BSS_SECTION_NAME,(NOLOAD),)
 {
 #ifdef CONFIG_USERSPACE

